server/stream.go

```diff
@@ -4638,7 +4646,9 @@ func (mset *stream) removeConsumer(o *consumer) {
 		}
 		// Always remove from the leader sublist.
 		if mset.csl != nil {
-			mset.csl.Remove(o.signalSub())
+			for _, sub := range o.signalSubs() {
```

Comment:
Rigjt now the performance would be affected only by number of filters, not consumers.
I adopted the behaviour when rebasing after this improvement landed.

You are right though - we should be able to have one callback per consumer and reduce number of callbacks significantly if someone uses few multiple filtered consumers instead of a lot of single-filtered ones.

Will change this one.